src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddFileMetadataToDeposit.scala

```diff
@@ -17,22 +17,50 @@ package nl.knaw.dans.easy.multideposit.actions
 
 import java.io.File
 import java.net.URLConnection
+import java.nio.file.Paths
 
 import nl.knaw.dans.easy.multideposit.actions.AddFileMetadataToDeposit._
 import nl.knaw.dans.easy.multideposit.{Action, Settings, _}
 import org.apache.commons.logging.LogFactory
 
-import scala.util.{Failure, Try}
+import scala.util.{Failure, Success, Try}
 
-case class AddFileMetadataToDeposit(row: Int, datasetID: DatasetID)(implicit settings: Settings) extends Action {
+case class AddFileMetadataToDeposit(row: Int, entry: (DatasetID, Dataset))(implicit settings: Settings) extends Action {
 
   val log = LogFactory.getLog(getClass)
 
+  val (datasetID, dataset) = entry
+
   def run() = {
     log.debug(s"Running $this")
 
     writeFileMetadataXml(row, datasetID)
   }
+
+  /**
+   * Verifies whether all preconditions are met for this specific action.
+   *
+   * @return `Success` when all preconditions are met, `Failure` otherwise
+   */
+  override def checkPreconditions: Try[Unit] = {
+    log.debug(s"Checking preconditions for $this")
+
+    val inputDir = settings.multidepositDir;//multiDepositDir(settings, datasetID)
+    val inputDirPath = Paths.get(inputDir.getAbsolutePath)
+
+    // Note that the FILE_SIP paths are not really used in this action
+    val nonExistingPaths = dataset.get("FILE_SIP").getOrElse(List.empty)
+      .filter(fp => fp.nonEmpty)
+      .filterNot(fp => {
```

Comment:
1. put this on one line. code block is not needed here.
2. either use the `Path` API or use the `File` API. To check `exists` with the `Path` API, use `Files.exists`. See also https://docs.oracle.com/javase/tutorial/essential/io/legacy.html.